cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/SecureStoreHandler.java

```diff
@@ -78,54 +68,43 @@ public void create(HttpRequest httpRequest, HttpResponder httpResponder, @PathPa
                                       " \n" + GSON.toJson(dummy));
     }
 
-    String description = secureKeyCreateRequest.getDescription();
-    String value = secureKeyCreateRequest.getData();
-    if (Strings.isNullOrEmpty(value)) {
-      throw new BadRequestException("The data field should not be empty. This is the data that will be stored " +
-                                      "securely.");
-    }
-
-    byte[] data = value.getBytes(StandardCharsets.UTF_8);
-    secureStoreManager.put(namespace, name, data, description, secureKeyCreateRequest.getProperties());
+    secureStoreService.put(secureKeyCreateRequest, secureKeyId);
     httpResponder.sendStatus(HttpResponseStatus.OK);
   }
 
   @Path("/keys/{key-name}")
   @DELETE
   public void delete(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("namespace-id") String namespace,
-                     @PathParam("key-name") String name) throws IOException {
-    secureStoreManager.delete(namespace, name);
+                     @PathParam("key-name") String name) throws Exception {
+    SecureKeyId secureKeyId = new SecureKeyId(namespace, name);
+    secureStoreService.delete(secureKeyId);
     httpResponder.sendStatus(HttpResponseStatus.OK);
   }
 
   @Path("/keys/{key-name}")
   @GET
   public void get(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("namespace-id") String namespace,
-                  @PathParam("key-name") String name) throws IOException {
-    SecureStoreData secureStoreData = secureStore.get(namespace, name);
-    String data = new String(secureStoreData.get(), StandardCharsets.UTF_8);
+                  @PathParam("key-name") String name) throws Exception {
+    SecureKeyId secureKeyId = new SecureKeyId(namespace, name);
+    String data = new String(secureStoreService.get(secureKeyId).get(), StandardCharsets.UTF_8);
     httpResponder.sendJson(HttpResponseStatus.OK, data);
   }
 
   @Path("/keys/{key-name}/metadata")
   @GET
   public void getMetadata(HttpRequest httpRequest, HttpResponder httpResponder,
                           @PathParam("namespace-id") String namespace, @PathParam("key-name") String name)
```

Comment:
nit: if you move the ``name`` parameter to the next line, indentation will be slightly better since the throws clause will move towards the right.